app/javascript/packages/transliterable-field-group/index.ts

```diff
@@ -0,0 +1,375 @@
+import { trackError } from '@18f/identity-analytics';
+import { request } from '@18f/identity-request';
+
+/**
+ * Representation of the response from a transliteration API call
+ */
+interface ValidationResponse {
+  success: boolean;
+  data: Record<string, string>;
+}
+
+/**
+ * Internal cache structure for controlling when API call is made
+ * and reusing result
+ */
+interface ValidationResultCache {
+  preValidationValues: Record<string, string>;
+  postValidationValues: Record<string, string>;
+  result: Record<string, string>;
+}
+
+/**
+ * This adds an asynchronous transliteration check for certain form fields
+ * that gets run between form validation and submission.
+ */
+class TransliterableFieldGroupElement extends HTMLElement {
+  private static readonly DEFAULT_INPUT_TIMEOUT_MS = 1500;
+
+  /**
+   * Cache to moderate API calls and allow reuse of the previous result
+   */
+  private validationResultCache: ValidationResultCache = {
+    preValidationValues: {},
+    postValidationValues: {},
+    result: {},
+  };
+
+  private lastRevalidationRequest: Promise<void>;
+
+  private lastFormSubmit: Promise<any> | null = null;
```

Comment:
can we change this to `lastFormSubmission`?

Reply:
Thanks, I'll rename this before moving the PR out of draft status. If I can't think of a more concise name, then I will use the one you suggested.